src/PhpSpec/Runner/Maintainer/CollaboratorsMaintainer.php

```diff
@@ -140,6 +130,9 @@ private function generateCollaborators(CollaboratorManager $collaborators, \Refl
                 if (($indexedClass = $this->getParameterTypeFromIndex($classRefl, $parameter))
                     || ($indexedClass = $this->getParameterTypeFromReflection($parameter))) {
                     $collaborator->beADoubleOf($indexedClass);
+                    if (null !== $indexedClass) {
```

Comment:
well, I fully agree with you that assignments inside conditions are much harder to read. But this is not the topic here. It does not change the fact that ``$indexedClass`` can never be null here